idaes/models/control/tests/test_antiwindup.py

```diff
@@ -0,0 +1,571 @@
+
+    tsim = 40
+
+    # Next create a model for the 0 to 5 sec time period
+    m_dynamic, solver = create_model(
+        steady_state=False,
+        time_set=[0, 2, tsim],
+        nfe=2,
+        calc_integ=True,
+        tee=True,
+        derivative_on_error=False,
+        initial_valve1_opening=s1_valve,
+        antiwindup=ControllerAntiwindupType.NONE,
+    )
+    # return m_dynamic, solver
+    # Retune controller to result in windup
+    m_dynamic.fs.ctrl.gain_p.fix(5e-6)
+    m_dynamic.fs.ctrl.gain_i.fix(1e-5)
+    m_dynamic.fs.ctrl.gain_d.fix(0)
+    # Add a step change in outlet setpoint
+    _add_setpoint_step(m_dynamic, time=m_dynamic.fs.time.at(3), value=3.0e5)
+
+    res = petsc.petsc_dae_by_time_element(
+        m_dynamic,
+        time=m_dynamic.fs.time,
+        between=[
+            m_dynamic.fs.time.first(),
+            m_dynamic.fs.time.at(2),
+            m_dynamic.fs.time.last(),
+        ],
+        ts_options={
+            "--ts_type": "beuler",  # Backwards Euler
+            "--ts_adapt_type": "basic",
+            "--ts_dt": 1,
+            "--ts_save_trajectory": 1,
+            # "--snes_monitor": "",
+            # "--ksp_monitor": "",
+            # "--ts_monitor": "",
+            "--ts_max_snes_failures": 1000,
+        },
+    )
+    # read the trajectory data, and make it easy by interpolating a time point
+    # every second
+    tj = res.trajectory
+    tj2 = tj.interpolate(np.linspace(0, tsim, tsim + 1))
+    tf = m_dynamic.fs.time.last()
+
+    # Test for huge value  of integral term and oscillatory control trajectory
+    assert tj2.get_vec(m_dynamic.fs.ctrl.mv_integral_component[tf])[8] >= 2.95
+    assert tj2.get_vec(m_dynamic.fs.valve_1.valve_opening[tf])[13] == pytest.approx(
+        1, abs=1e-4
+    )
+    assert tj2.get_vec(m_dynamic.fs.valve_1.valve_opening[tf])[16] <= 0.32
+    assert tj2.get_vec(m_dynamic.fs.valve_1.valve_opening[tf])[19] >= 0.82
+    assert tj2.get_vec(m_dynamic.fs.valve_1.valve_opening[tf])[23] <= 0.56
+
+    return m_dynamic, solver, tj2
+
+
+@pytest.mark.skipif(not petsc.petsc_available(), reason="PETSc solver not available")
+@pytest.mark.component
+def test_setpoint_change_conditional_integration():
+    # First calculate the two steady states that should be achieved in the test
+    # don't worry these steady state problems solve super fast
+    m_steady, solver = create_model(tee=False)
+    m_steady.fs.valve_1.inlet.pressure.fix(6.0e5)
+    m_steady.fs.tank_2.control_volume.properties_out[0].pressure.fix(1.5e5)
+    m_steady.fs.valve_1.valve_opening[0].unfix()
+    solver.solve(m_steady, tee=False)
+    s1_valve = pyo.value(m_steady.fs.valve_1.valve_opening[0])
+    print(f"Steady state 1 valve opening: {s1_valve}")
+
+    m_steady.fs.tank_2.control_volume.properties_out[0].pressure.fix(3.0e5)
+    solver.solve(m_steady, tee=False)
+    s2_valve = pyo.value(m_steady.fs.valve_1.valve_opening[0])
+    print(f"Steady state 2 valve opening: {s2_valve}")
```

Comment:
This code is repeated a few times. Can the steady state solves be moved to `create_model`?